python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -800,13 +803,47 @@ def get_direct_input_values_from_job(target: PipelineDefinition) -> Mapping[str,
 @io_manager(
     description="Built-in filesystem IO manager that stores and retrieves values using pickling."
 )
-def default_job_io_manager(init_context: "InitResourceContext"):
+def _default_job_io_manager(init_context: "InitResourceContext"):
     from dagster._core.storage.fs_io_manager import PickledObjectFilesystemIOManager
 
     instance = check.not_none(init_context.instance)
     return PickledObjectFilesystemIOManager(base_dir=instance.storage_directory())
 
 
+def get_default_job_io_manager():
+    io_manager_env_var = os.getenv("DEFAULT_IO_MANAGER_DEFINITION_ATTRIBUTE")
```

Comment:
I put together a kind of fun proof-of-concept for this type of approach while out:

https://github.com/jmsanders/depenv

The ergonomics potentially leave something to be desired although I like your idea in this PR of using a json string.

But in general, I agree with @sryza re: the slippery slope. It feels odd to me that you could set IO managers this way but not other resources. And I'm not quite sure I'm ready to fully advocate for setting everything this way without dogfooding the idea a bit more.